frontend/src/views/components/Header.tsx

```diff
@@ -30,6 +32,14 @@ export const Header = () => {
     dispatch(toggleTheme());
   };
 
+  useEffect(() => {
+    const fetchFirstName = async () => {
+      const userInfo = await UserService.getUserInfoFromSession();
+      if (userInfo) dispatch(login(userInfo.user_metadata.firstName));
+    };
+    fetchFirstName();
+  }, [dispatch]);
```

Comment:
If it only refers to the value from local storage, I couldn't figure out how to fetch the data again when the local storage data is updated. Header component is not parent-child relationships with each component and there's no rendering timing when the pages are rendered.